app/src/main/java/com/zeapo/pwdstore/ui/dialogs/FolderCreationDialogFragment.kt

```diff
@@ -5,41 +5,95 @@
+import androidx.appcompat.app.AppCompatActivity
 import androidx.core.os.bundleOf
 import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.checkbox.MaterialCheckBox
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
 import com.zeapo.pwdstore.PasswordStore
 import com.zeapo.pwdstore.R
+import com.zeapo.pwdstore.crypto.BasePgpActivity
+import com.zeapo.pwdstore.crypto.GetKeyIdsActivity
+import com.zeapo.pwdstore.utils.PasswordRepository
+import com.zeapo.pwdstore.utils.PasswordRepository.Companion.getRepositoryDirectory
+import com.zeapo.pwdstore.utils.commitChange
 import com.zeapo.pwdstore.utils.requestInputFocusOnView
 import java.io.File
+import kotlinx.coroutines.launch
+import me.msfjarvis.openpgpktx.util.OpenPgpApi
 
 class FolderCreationDialogFragment : DialogFragment() {
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
         val alertDialogBuilder = MaterialAlertDialogBuilder(requireContext())
         alertDialogBuilder.setTitle(R.string.title_create_folder)
         alertDialogBuilder.setView(R.layout.folder_dialog_fragment)
-        alertDialogBuilder.setPositiveButton(getString(R.string.button_create)) { _, _ ->
-            createDirectory(requireArguments().getString(CURRENT_DIR_EXTRA)!!)
-        }
+        alertDialogBuilder.setPositiveButton(getString(R.string.button_create), null)
         alertDialogBuilder.setNegativeButton(getString(android.R.string.cancel)) { _, _ ->
             dismiss()
         }
         val dialog = alertDialogBuilder.create()
         dialog.requestInputFocusOnView<TextInputEditText>(R.id.folder_name_text)
+        dialog.setOnShowListener {
+            dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener {
+                createDirectory(requireArguments().getString(CURRENT_DIR_EXTRA)!!)
+            }
+        }
         return dialog
     }
 
     private fun createDirectory(currentDir: String) {
         val dialog = requireDialog()
-        val materialTextView = dialog.findViewById<TextInputEditText>(R.id.folder_name_text)
-        val folderName = materialTextView.text.toString()
-        val newFolder = File("$currentDir/$folderName")
+        val folderNameView = dialog.findViewById<TextInputEditText>(R.id.folder_name_text)
+        val folderNameViewContainer = dialog.findViewById<TextInputLayout>(R.id.folder_name_container)
+        val newFolder = File("$currentDir/${folderNameView.text}")
+        if (newFolder.exists()) {
+            folderNameViewContainer.error = when {
+                newFolder.isFile -> getString(R.string.folder_creation_err_file_exists)
+                newFolder.isDirectory -> getString(R.string.folder_creation_err_folder_exists)
+                // Should be impossible but eh.
+                else -> return
```

Comment:
Since `isFile` and `isDirectory` check `exists()` anyway, all that is needed is this `when` and not the `if` around it.